src/connector.js

```diff
@@ -39,7 +40,8 @@ class Connector {
   }
 
   executeForHostname(cb) {
-    dns.lookup(this.options.host, { all: true }, (err, addresses) => {
+    const serverName = this.options.serverNameAsACE ? (punycode.toASCII(this.options.host)).trim() : this.options.host;
```

Comment:
Are you sure the `trim` call is required here?

Reply:
Yes, it is needed. We have couple of IDN servers internally, using the value of `punycode.toASCII()` in `dns.lookup` worked for few IDN servers and not for all, punycode conversion turns out to have trailing spaces 😅 
I will test its need again after switching to bestiejs/punycode.js 👍

Reply:
> turns out that punycode conversion has trailing spaces 😅

Wow, that seems like a pretty serious bug? :O Do you have an example of such a domain name?

Reply:
My apologies 😢Just went back and did a double-triple check, problem was in `config.server`, that had a trailing spaces, so of course the punycode had it to.
Thanks @arthurschreiber 😌